src/animals/Giraffe.ts

```diff
@@ -16,20 +17,21 @@ export default class Giraffe extends Mammal implements ICanRun {
   ) {
     super(name, age, isMale);
     if (height <= 0) {
-      throw Error("Elephant height must be greater than zero!")
+      throw Error(GIRAFFE_HEIGHT_NEGATIVE)
     };
     this._canHaveFamily = true;
     this.height = height;
   };
 
-  run(): void {
-    console.log(`${this.name} is running`);
+  run(): string {
+    return `${this.name} is running`;
   };
 
 
-  public giveBirth(): void {
+  public giveBirth() {
     const family = families.find(giraffe => giraffe.name == this.home);
     const candidateGiraffes: AnimalCandidate[] = this.generateBabyProperties(this.numberOfBabiesBorn);
+    const babies: Giraffe[] = [];
     if (candidateGiraffes) {
```

Comment:
i think there is no need for an "if" here, is there a chance that method "generateBabyProperties  does not return? . Anyway you can just do candidateGiraffes?.forEach...
Even better you can  try "return candidateGiraffes?.map(candidate => {
_some code here_
return babyGiraffe
}